File: csharp/1078.occurrences-after-bigram.cs
/*
 * @lc app=leetcode id=1078 lang=csharp
 *
 * [1078] Occurrences After Bigram
 */

using System.Collections.Generic;

// @lc code=start
public class Solution {
    public string[] FindOcurrences(string text, string first, string second) {
        var split = text.Split(' ');
        int n = split.Length;
        var ret = new List<string>();
        for (int i = 0; i + 2 < n; ++i) {
            if (split[i] == first && split[i + 1] == second) {
                ret.Add(split[i + 2]);
            }
        }
        return ret.ToArray();
    }
}
// @lc code=end

File: java/1078.occurrences-after-bigram.java
/*
 * @lc app=leetcode id=1078 lang=java
 *
 * [1078] Occurrences After Bigram
 */

import java.util.ArrayList;

// @lc code=start
import java.util.regex.Pattern;

class Solution {
    public String[] findOcurrences(String text, String first, String second) {
        var pattern = Pattern.compile("(?=((^| )%s %s (\\w+)($| )))".formatted(first, second));
        var matcher = pattern.matcher(text);
        var ret = new ArrayList<String>();
        matcher.results().forEach(result -> ret.add(result.group(3)));
        return ret.toArray(String[]::new);
    }
}
// @lc code=end

